src/main/java/com/example/onlinebookstore/service/impl/CartServiceImpl.java

```diff
@@ -0,0 +1,65 @@
+package com.example.onlinebookstore.service.impl;
+
+import com.example.onlinebookstore.dto.CartItemDto;
+import com.example.onlinebookstore.dto.CreateCartItemRequestDto;
+import com.example.onlinebookstore.dto.ShoppingCartDto;
+import com.example.onlinebookstore.dto.UpdateCartItemRequestDto;
+import com.example.onlinebookstore.exception.EntityNotFoundException;
+import com.example.onlinebookstore.mapper.ShoppingCartMapper;
+import com.example.onlinebookstore.model.CartItem;
+import com.example.onlinebookstore.model.ShoppingCart;
+import com.example.onlinebookstore.repository.CartItemRepository;
+import com.example.onlinebookstore.repository.ShoppingCartRepository;
+import com.example.onlinebookstore.repository.book.BookRepository;
+import com.example.onlinebookstore.repository.user.UserRepository;
+import com.example.onlinebookstore.service.CartService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Service;
+
+@RequiredArgsConstructor
+@Service
+public class CartServiceImpl implements CartService {
+    private final BookRepository bookRepository;
+    private final CartItemRepository cartItemRepository;
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final ShoppingCartMapper cartMapper;
+    private final UserRepository userRepository;
+
+    @Override
+    public CartItemDto save(String email, CreateCartItemRequestDto requestDto) {
+        CartItem cartItem = cartMapper.toCartItemModel(requestDto);
```

Comment:
let's make your code more readable
Create Book book = repo.find
ShoppingCart cart = repo.find
cart.getCartItems() -> add new cartItem, check if there is already bookId exists then increase quantity
if no -> add new cartItem